Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -0,0 +1,163 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+
+namespace Content.Server.GameTicking.Rules;
+
+public sealed class NukeopsRuleSystem : GameRuleSystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly IMapLoader _mapLoader = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly StationSpawningSystem _stationSpawningSystem = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly RoundEndSystem _roundEndSystem = default!;
+
+    private Dictionary<Mind.Mind, bool> _aliveNukeops = new();
+    private bool _opsWon;
+
+    public override string Prototype => "Nukeops";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RoundStartAttemptEvent>(OnStartAttempt);
+        SubscribeLocalEvent<RulePlayerSpawningEvent>(OnPlayersSpawning);
+        SubscribeLocalEvent<MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<RoundEndTextAppendEvent>(OnRoundEndText);
+        SubscribeAllEvent<NukeExplodedEvent>(OnNukeExploded);
+    }
+
+    private void OnNukeExploded(NukeExplodedEvent ev)
+    {
+        _opsWon = true;
+        _roundEndSystem.EndRound();
+    }
+
+    private void OnRoundEndText(RoundEndTextAppendEvent ev)
+    {
+        ev.AddLine(_opsWon ? Loc.GetString("nukeops-ops-won") : Loc.GetString("nukeops-crew-won"));
+        ev.AddLine(Loc.GetString("nukeops-list-start"));
+        foreach (var nukeop in _aliveNukeops)
+        {
+            ev.AddLine($"- {nukeop.Key.Session?.Name}");
+        }
+    }
+
+    private void OnMobStateChanged(MobStateChangedEvent ev)
+    {
+        if (!_aliveNukeops.TryFirstOrNull(x => x.Key.OwnedEntity == ev.Entity, out var op)) return;
+
+        _aliveNukeops[op.Value.Key] = op.Value.Key.CharacterDeadIC;
+
+        if (_aliveNukeops.Values.All(x => !x))
+        {
+            _roundEndSystem.EndRound();
+        }
+    }
+
+    private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
+    {
+        _aliveNukeops.Clear();
+
+        var numOps = (int)Math.Min(Math.Floor((double)ev.PlayerPool.Count / _cfg.GetCVar(CCVars.NukeopsPlayersPerOp)),
+            _cfg.GetCVar(CCVars.NukeopsMaxOps));
+        var ops = new IPlayerSession[numOps];
+        for (var i = 0; i < numOps; i++)
+        {
+            ops[i] = _random.PickAndTake(ev.PlayerPool);
+        }
+
+        var map = "/Maps/syndiepuddle.yml";
+
+        var aabbs = _stationSystem.Stations.SelectMany(x =>
+            Comp<StationDataComponent>(x).Grids.Select(x => _mapManager.GetGridComp(x).Grid.WorldBounds)).ToArray();
```

Comment:
wdym all grids? on the map in general? this would mean that an outlier tile floating in -1000 -1000 would set back the shuttlespawn significantly.

as for checking that the spawnarea is clear: ye, i agree, but how? :P

Reply:
I think salvage needs a redesign anyway because it's small af but for now probably just expand the danger zone.